hyperspy/drawing/spectrum.py

```diff
@@ -331,7 +331,16 @@ def update(self, force_replot=False):
             ydata = self.data_function(axes_manager=self.axes_manager).real
         else:
             ydata = self.data_function(axes_manager=self.axes_manager).imag
-        self.line.set_ydata(ydata)
+
+        new_axis = self.axes_manager.signal_axes[0].axis
```

Comment:
Anyways, this code seems like a bit of a hack around a bigger issue: Should `update()` calls refresh the axes of the plot? From MPL's point of view it is possible, but the way HE/HSE/HIE is designed doesn't really support this.